extension/wallet/src/storage.rs

```diff
@@ -10,14 +11,11 @@ pub struct Storage {
 }
 
 impl Storage {
-    pub async fn get_open_loans(&self) -> Result<Vec<LoanDetails>> {
-        let loans = match self
-            .get_item::<String>("open_loans")
-            .context("no key \"open_loans\" in local storage")?
-        {
-            Some(loans) => serde_json::from_str(&loans)?,
-            None => Vec::<LoanDetails>::new(),
-        };
+    pub fn get_open_loans(&self) -> Result<Vec<LoanDetails>> {
+        let loans = self
+            .get_json_item::<Vec<LoanDetails>>("open_loans")
+            .context("no key 'open_loans' in local storage")?
```

Comment:
I think that error message is not quite right. The result here represents deserialization errors.

The empty case is handled with the `unwrap_or_default`.

Reply:
good catch.